graylog2-server/src/main/java/org/graylog2/periodical/ESVersionCheckPeriodical.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.periodical;
+
+import org.graylog2.notifications.Notification;
+import org.graylog2.notifications.NotificationService;
+import org.graylog2.plugin.Version;
+import org.graylog2.plugin.periodical.Periodical;
+import org.graylog2.storage.ElasticsearchVersion;
+import org.graylog2.storage.versionprobe.VersionProbe;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+
+public class ESVersionCheckPeriodical extends Periodical {
+    private static final Logger LOG = LoggerFactory.getLogger(ESVersionCheckPeriodical.class);
+    private final Version initialElasticsearchVersion;
+    private final Optional<Version> versionOverride;
+    private final List<URI> elasticsearchHosts;
+    private final VersionProbe versionProbe;
+    private final NotificationService notificationService;
+
+    @Inject
+    public ESVersionCheckPeriodical(@ElasticsearchVersion Version elasticsearchVersion,
+                                    @Named("elasticsearch_version") @Nullable Version versionOverride,
+                                    @Named("elasticsearch_hosts") List<URI> elasticsearchHosts,
+                                    VersionProbe versionProbe,
+                                    NotificationService notificationService) {
+        this.initialElasticsearchVersion = elasticsearchVersion;
+        this.versionOverride = Optional.ofNullable(versionOverride);
+        this.elasticsearchHosts = elasticsearchHosts;
+        this.versionProbe = versionProbe;
+        this.notificationService = notificationService;
+    }
+
+    @Override
+    public boolean runsForever() {
+        return false;
+    }
+
+    @Override
+    public boolean stopOnGracefulShutdown() {
+        return true;
+    }
+
+    @Override
+    public boolean masterOnly() {
+        return true;
+    }
+
+    @Override
+    public boolean startOnThisNode() {
+        return true;
+    }
+
+    @Override
+    public boolean isDaemon() {
+        return false;
```

Comment:
I think this can run in the daemon thread pool. What do you think?

Reply:
Good point!